crates/re_ui/src/section_collapsing_header.rs

```diff
@@ -0,0 +1,172 @@
+    help: Option<Box<dyn FnOnce(&mut egui::Ui) + 'a>>,
+}
+
+impl<'a> SectionCollapsingHeader<'a> {
+    /// Create a new [`Self`].
+    ///
+    /// See also [`crate::UiExt::section_collapsing_header`]
+    pub fn new(label: impl Into<egui::WidgetText>) -> Self {
+        Self {
+            label: label.into(),
+            default_open: true,
+            button: None,
+            help: None,
+        }
+    }
+
+    /// Set the default open state of the section header.
+    ///
+    /// Defaults to `true`.
+    #[inline]
+    pub fn default_open(mut self, default_open: bool) -> Self {
+        self.default_open = default_open;
+        self
+    }
+
+    /// Set the button to be shown in the header.
+    #[inline]
+    pub fn button(mut self, button: HeaderMenuButton<'a>) -> Self {
+        self.button = Some(button);
+        self
+    }
+
+    /// Set the help text tooltip to be shown in the header.
+    #[inline]
+    pub fn help_text(mut self, help: impl Into<egui::WidgetText>) -> Self {
+        let help = help.into();
+        self.help = Some(Box::new(move |ui| {
+            ui.label(help);
+        }));
+        self
+    }
+
+    /// Set the help UI closure to be shown in the header.
+    #[inline]
+    pub fn help_ui(mut self, help: impl FnOnce(&mut egui::Ui) + 'a) -> Self {
+        self.help = Some(Box::new(help));
+        self
+    }
+
+    /// Display the header.
+    pub fn show(
+        self,
+        ui: &mut egui::Ui,
+        add_body: impl FnOnce(&mut egui::Ui),
+    ) -> egui::CollapsingResponse<()> {
+        let Self {
+            label,
+            default_open,
+            button,
+            help,
+        } = self;
+
+        let id = ui.make_persistent_id(label.text());
+
+        let mut content = list_item::LabelContent::new(label);
+        if button.is_some() || help.is_some() {
+            content = content
+                .with_buttons(|ui| {
+                    let button_response = button.map(|button| button.show(ui));
+                    let help_response = help.map(|help| ui.help_hover_button().on_hover_ui(help));
+
+                    match (button_response, help_response) {
+                        (Some(button_response), Some(help_response)) => {
+                            button_response | help_response
+                        }
+                        (Some(response), None) | (None, Some(response)) => response,
+                        (None, None) => unreachable!("at least one of button or help is set"),
+                    }
+                })
+                .always_show_buttons(true);
+        }
+
+        let resp = list_item::ListItem::new()
+            .interactive(true)
+            .force_background(DesignTokens::section_collapsing_header_color())
+            .show_hierarchical_with_children_unindented(ui, id, default_open, content, |ui| {
+                //TODO(ab): this space is not desirable when the content actually is list items
```

Comment:
probably link to the list itemification ticket? Once everything is linst items it should go away

On that note: Now that this is a list item, does it mean that we can remove some of the list item scopes we add down the line?

Reply:
I'm tempted to get rid of these right now, and add them where we dont have ListItem. I have that anyway on my TODO (#6655).

I can't quite answer for list item scope now, I have fully tracked the recent proliferation :)